openmrs/src/main/java/org/motechproject/openmrs/resource/impl/BaseResource.java

```diff
@@ -85,6 +88,9 @@ protected void delete(Config config, String path, Object... params) {
 
     private ResponseEntity<String> exchange(Config config, URI url, HttpMethod method, String body, HttpHeaders headers) {
         headers.add("Authorization", "Basic " + prepareCredentials(config));
+
+        httpClient.getState().clearCookies();
```

Comment:
I think better idea will be remove cookie header from request or exclude using cookies by httpClient during initialization. Please do some research in this direction